### PR TITLE
Multiple Set-Cookie headers should not be overwritten

### PR DESCRIPTION
**Pull request by Skn0tt**
When setting multiple headers, currently all but the last one are overwritten. This is prohibitive especially for using `Set-Cookie`, where multiple cookies are sent as multiple headers. This PR provides a fix for the bug, and sets up unit testing using Vitest to prove that the fix works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
   "scripts": {
     "prepack": "npm run clean && npm run compile",
     "compile": "tsc --build tsconfig.json",
-    "clean": "rm -rf dist"
+    "clean": "rm -rf dist",
+    "test": "vitest run"
```
Comment (Skn0tt) [inline package.json:15]: Happy to swap this out for a different test runner, or remove it altogether if you prefer that!

File: src/http-compute-js/http-outgoing.ts [hunk 2]
```diff
@@ -155,7 +155,7 @@ export class WrittenDataBuffer {
 export type HeadersSentEvent = {
   statusCode: number,
   statusMessage: string,
-  headers: Record<string, string>,
+  headers: [header: string, value: string][],
```
Comment (Skn0tt) [inline src/http-compute-js/http-outgoing.ts:158]: This is the main change. We can also change this to use `OutgoingHttpHeaders` if that works better!

**Review by harmony7 — APPROVED** [review]
Thank you for this, and thank you for even providing tests.  vitest is fine for now, we might add more tests just using Node.js native test runner later.

**Comment by harmony7** [comment]
released 1.1.4

**Pull request merged**